api/src/main/java/io/grpc/Metadata.java

```diff
@@ -118,34 +122,80 @@ public String parseAsciiString(String serialized) {
    * Constructor called by the transport layer when it receives binary metadata. Metadata will
    * mutate the passed in array.
    *
-   * @param usedNames the number of
+   * @param usedNames the number of names
    */
   Metadata(int usedNames, byte[]... binaryValues) {
     assert (binaryValues.length & 1) == 0 : "Odd number of key-value pairs " + binaryValues.length;
     size = usedNames;
     namesAndValues = binaryValues;
   }
 
-  private byte[][] namesAndValues;
+  /**
```

Comment:
This above constructor can reuse this constructor.

```java
  Metadata(int usedNames, byte[]... binaryValues) {
    this(usedNames, (Object[]) binaryValues);
  }
```

Reply:
Done.